text/0000-throw-expr.md

```diff
@@ -0,0 +1,908 @@
+- Feature Name: `throw_expr`
+- Start Date: 2018-04-30
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Introduce diverging `throw expr` expressions, typed at `!`,
+which either `break` to the closest `try { .. }` if there is one, or if not,
+`return` from the `fn` or closure. The expression form `throw expr` is supported
+on edition 2018 onwards. This also means that `throw` is reserved as a keyword.
+
+# Motivation
+[motivation]: #motivation
+
+## Highlighting the *"unhappy path"*
+
+[RFC 243]: https://github.com/rust-lang/rfcs/blob/master/text/0243-trait-based-exception-handling.md#-operator
+
+[RFC 243] writes regarding the `?` operator that:
+
+> The `?` operator itself is suggestive, syntactically lightweight enough
+> to not be bothersome, and lets the reader determine at a glance where an
+> exception may or may not be thrown.
+
+This RFC takes the moral of that story to heart and agrees that you should be
+able to *"determine at a glance where an exception may or may not be thrown"*.
+However, currently, when you write logic such as:
+
+```rust
```

Comment:
Good catch; However I think (hope) people will see beyond that :)

Reply:
The highlighting appears to be correct on everything in the examples other than `throw` itself, including the expr being thrown, so I think we're getting a much more accurate impression already than we would with no highlighting at all.

Reply:
> I think (hope) people will see beyond that

I agree that people can see beyond it. I just know that I at least have a hard to ignore subconcious reaction to the inconsistent highlighting on

```rust
if condition { throw Foo } else { return Ok(Bar) }
```

which makes it more difficult for me to really judge the new syntax. I'm expecting that `throw` is the start of a normal expression since it's not a keyword. Whereas I find it much easier to suppress that and treat `throw` as a keyword without any highlighting

```
if condition { throw Foo } else { return Ok(Bar) }
```